TEMP: Switch to a patched CloudflareSolverRe package.

thailyn opened the pull request:
This package is a fork of the "official" CloudflareSolverRe, but it includes a fix for a Cloudflare change that was just introduced in the last day or so.

We should probably not merge this PR, and instead wait for the package we are using already to merge this patch.  In the meantime, however, this PR can provide a temporary working version of Procurement.

This PR also updates the Procurement projects to the .NET Framework 4.6.2 (from 4.5).  Newer versions of the CloudflareSolverRe package requires the .NET Framework 4.6.

**EDIT:** The latest version of this patch is from **2020.05.28**.  It can be downloaded [here](https://ci.appveyor.com/api/buildjobs/pcge9o9kjff03knt/artifacts/Procurement%2Fbin%2FProcurement.zip).  This download link is available for six months, until 2020.11.28.

thailyn committed:
Switch to a patched CloudflareSolverRe package.

This package is a fork of the "official" CloudflareSolverRe, but it includes a fix for a Cloudflare change that was just introduced in the last day or so.

thailyn committed:
Update Cloudflare library to version 1.0.6.2.

ifkvothe commented:
Hello, I confirm positive income, thank so much!!!

thailyn committed:
Update target framework from 4.5 to 4.6.2.

thailyn committed:
Update CloudflareSolverRe from 1.0.6.2 to 1.0.7.

This latest version of the Ngosang fork of CloudflareSolverRe requires
.NETStandard 1.3, potentially because of a new transitive dependency
on Jint.  The new version of this package required adding several new
packages, and there are potential version conflicts that need to be
resolved.  However, this does not prevent the project from building
and appearing to run correctly.

thailyn committed:
Add app.config files generated after with 4.6.2.

It appears these files are now created for each project, after moving
to targetting .NET Framework 4.6.2.

thailyn commented:
> 
> 
> hi, crash is procurement appI am lost, that is, should I modify the .dll files?

I don't really understand what the issue with Procurement is that you're experiencing.  Please file a new issue and include more details, and the `DebugInfo.log` file.

ifkvothe commented:
> > hola, el bloqueo es la aplicación de adquisición Estoy perdido, es decir, ¿debería modificar los archivos .dll?
> 
> Realmente no entiendo cuál es el problema con Adquisiciones que estás experimentando. Presente un nuevo problema e incluya más detalles y el `DebugInfo.log`archivo.


the application does not want to open me. I have tried to enter a new property but it has not been successful

[25-04-2020 8:29] Failed to build HTTP request and get response for: method = GET, url = 'https: //www.pathofexile.com/login', allowAutoRedirects =, requestData = '': Error in the remote server: (503) Server not available.

thailyn committed:
Retarget NuGet packages to 4.6.2 from 4.5.

These packages are not automatically updated, so this was accomplished
by executing the command "Update-Package -Reinstall" in the Package
Manager Console.

thailyn committed:
Remove some references to System.Net.Http.

The POEApi.TestHelpers and Procurement.Tests projects were including
this assembly at version 4.0.0, but other deependencies required
4.1.0.  Simply removing this reference from those two projects appear
to be sufficient to fix the version conflicts that were reported
during building.

ashneil12 commented:
I'm getting [Error] One or more errors occurred. when trying to log in using this one.

thailyn committed:
Update CloudflareSolverReNgosang to 1.0.7.2.

thailyn commented:
@ashneil12 Sorry for the delay.  Please use the most recent version, released today, linked to from the initial description of this PR.

lstefano71 commented:
Close, but no cigar. OK, with Cloudflare, but I get: "[Error] Could not find a part of the path '...\poe-procurement-fork\wildy2k\characterdata.bin'."

thailyn commented:
No worries, that's a bug that occasionally surfaces that I haven't been able to track down the root cause.  Luckily, it seems transient and Procurement usually works when run again.

ZeldaZocker commented:
Hey, first of all thx for the fixed version. Sadly it doesn't work for me anymore. Im now getting Error 503 again. [Here](https://github.com/Stickymaddness/Procurement/files/4691800/DebugInfo.log) is the DebugInfo.log

thailyn commented:
Thanks for letting us know.  I don't see much discussion happening in the Cloudflare library we use about Cloudflare updates, but there is a new version of the library, so I will see if that fixes Procurement.

Motecra commented:
hello, i still new about this so what file name must i change and how to change it?

enki001 commented:
hi, same problem here, seems like every week the change somenthing! i get Error, ty

thailyn committed:
Update CloudflareSolverReNgosang to 1.0.7.3.

thailyn commented:
I've updated this PR to use the latest version of the Cloudflare library that Procurement depends on.  I was able to replicate the login issue, and this appears to fix it.  Please use the download link in the initial description for this PR, and let me know if there are any issues.  Thanks!

Motecra commented:
thanks, after download it working now for me

enki001 commented:
everything work ok now, ty

ifkvothe commented:
EDIT: The latest version of this patch is from 2020.05.28. It can be downloaded here. This download link is available for six months, until 2020.11.28. 
I confirm positive income!!! thanks you a lot!!!
